backend/src/main/java/gov/cdc/usds/simplereport/config/GraphQlConfig.java

```diff
@@ -0,0 +1,86 @@
+import graphql.GraphQLError;
+import graphql.validation.rules.OnValidationErrorStrategy;
+import graphql.validation.rules.ValidationRules;
+import graphql.validation.schemawiring.ValidationSchemaWiring;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.graphql.execution.DataFetcherExceptionResolver;
+import org.springframework.graphql.execution.RuntimeWiringConfigurer;
+import org.springframework.security.access.AccessDeniedException;
+import reactor.core.publisher.Mono;
+
+@Slf4j
+@Configuration
+public class GraphQlConfig {
+  public static final String REQUIRED_PERMISSIONS_DIRECTIVE_NAME = "requiredPermissions";
+  public static final int MAXIMUM_SIZE = 256;
+
+  // mask all exception by default
+  // to customize errors returned, you have to manually throw it here
+  @Bean
+  public DataFetcherExceptionResolver dataFetcherExceptionResolver() {
+    return (exception, environment) -> {
+      String errorPath = environment.getExecutionStepInfo().getPath().getSegmentName();
+
+      log.error("Will replace the following exception with a GenericGraphqlException: ", exception);
+
+      if (exception instanceof ConflictingUserException) {
+        return Mono.just(singletonList((GraphQLError) exception));
+      }
+
+      if (exception instanceof AccessDeniedException) {
+        return Mono.just(singletonList(new GenericGraphqlException("Unauthorized", errorPath)));
+      }
+
+      if (exception instanceof NonexistentUserException) {
+        return Mono.just(
+            singletonList(new GenericGraphqlException("Cannot find user.", errorPath)));
+      }
+
+      if (exception instanceof IllegalGraphqlArgumentException) {
+        return Mono.just(
+            singletonList(new GenericGraphqlException(exception.getMessage(), errorPath)));
+      }
+
+      if (exception instanceof IllegalGraphqlFieldAccessException) {
+        return Mono.just(
+            singletonList(new GenericGraphqlException(exception.getMessage(), errorPath)));
+      }
+
+      return Mono.just(singletonList(new GenericGraphqlException((errorPath))));
+    };
+  }
+
+  @Bean
+  public RuntimeWiringConfigurer runtimeWiringConfigurer() {
+    // this adds the graphql-java-extended-validation and a max argument size of MAXIMUM_SIZE
+    ValidationRules validationRules =
+        ValidationRules.newValidationRules()
+            .addRule(new DefaultArgumentValidation(MAXIMUM_SIZE))
+            .onValidationErrorStrategy(OnValidationErrorStrategy.RETURN_NULL)
+            .build();
+
+    return builder ->
+        builder
+            .scalar(UploadScalarType.upload)
+            .scalar(LocalDateScalar.LocalDate)
+            .scalar(DateTimeScalar.DateTime)
+            .directiveWiring(new RequiredPermissionsDirectiveWiring())
+            .directiveWiring(new ValidationSchemaWiring(validationRules));
```

Comment:
replaces [this code](https://github.com/CDCgov/prime-simplereport/pull/4243/files#diff-c9d7580fa2d627bb5d003ddb42b17bc3fff15e2ec21dcd1788a6c743921736ceL24)